### ISSUE TITLE
Allow a custom renderer for the stage

### ISSUE THREAD

**Issue by weavejester** [Izzimach/react-pixi#44]
It would be nice if the stage component had a `renderer` option where I could pass my own renderer to it. This would allow me to use the stage renderer in [RenderTexture](https://pixijs.github.io/docs/PIXI.RenderTexture.html) objects.

**Comment by Izzimach** [comment]
Well I can add a `renderer` prop that gets called instead of the default, but the `<canvas>` element would still be there. Do you want to totally replace the current renderer & canvas or just add some intermediate/postprocessing sort of thing?

**Comment by Izzimach** [comment]
Hm, I'm assuming that you take the `RenderTexture` and draw/store/send it somewhere, so the `PIXIStage` would just return a null element instead of a `<canvas>`? That sounds like what you want.

Since pixi 3.0 did away with the `Stage` I guess long term that `PIXIStage` should be replaced with one or more `Renderer` components though.

**Comment by Izzimach** [comment]
Dunno if it works yet, still needs a test.

**Comment by weavejester** [comment]
I wanted a pixelated look, so I was going to render the scene to a `RenderTexture`, then scale it up and apply it to the stage. Just scaling the stage produces some odd results, since any filters applied to sprites within the stage aren't scaled correspondingly.

I also found that the `cacheAsBitmap` option on containers had some odd behaviour under React, so I was going to use `RenderTexture` as a caching mechanism as well. Exposing the renderer seemed like a simple solution to the problem.

However... since then I've written some more code, and I think what I want to do is unusual enough not to be a good fit for React-Pixi. I'm using ClojureScript as well, so my use-case is doubly unusual! After some thought I've decided that I probably need to write my own Pixi wrapper library.

Thanks for your help, and I'll still keep an eye on React-Pixi for future projects :smile:
